### PR TITLE
Implementing consume_*() methods on Bigtable PartialRowsData.

### PR DESCRIPTION
**Pull request by dhermes**
These methods read from a stream of `ReadRowsResponse`'s, parse them and then store them on the `PartialRowsData` instance.

@tseaver The implementation of `PartialRowData.update_from_read_rows` is coming soon. It depends on #1492. I held off on including it there because it has a non-trivial amount of code.

### CODE REVIEW

File: gcloud/bigtable/row_data.py [hunk 1]
```diff
@@ -147,6 +147,19 @@ def clear(self):
...
+        :type read_rows_response_pb:
+            :class:`._generated.bigtable_service_messages_pb2.ReadRowsResponse`
+        :param read_rows_response_pb: A response streamed back as part of a
+                                      ``ReadRows`` request.
+
+        :raises: :class:`NotImplementedError <exceptions.NotImplementedError>`
+                 always
+        """
+        raise NotImplementedError
+
```
Comment (tseaver) [inline gcloud/bigtable/row_data.py]: Do we need a message explaining why we can't implement this method, or indicating what the caller should do instead?

File: gcloud/bigtable/row_data.py [hunk 2]
```diff
@@ -180,3 +193,45 @@ def rows(self):
...
+        :raises: :class:`StopIteration <exceptions.StopIteration>` if the
+                 response iterator has no more responses to stream.
+        """
+        read_rows_response = self._response_iterator.next()
+        row_key = read_rows_response.row_key
+        partial_row = self._rows.get(row_key)
+        if partial_row is None:
+            partial_row = self._rows[row_key] = PartialRowData(row_key)
+        # NOTE: This is not atomic in the case of failures.
+        partial_row.update_from_read_rows(read_rows_response)
```
Comment (tseaver) [inline gcloud/bigtable/row_data.py]: Is this not the method above which raises `NotImplementedError`?

File: gcloud/bigtable/row_data.py (hunk 1 above)
Comment (dhermes) [inline gcloud/bigtable/row_data.py]: See my initial [PR comment][1].

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/pull/1493#issue-134682820

File: gcloud/bigtable/row_data.py (hunk 2 above)
Comment (dhermes) [inline gcloud/bigtable/row_data.py]: Ditto.

Comment (tseaver) [inline gcloud/bigtable/row_data.py]: How can we be testing rows -> row interaction without the implementation of that method?

Comment (dhermes) [inline gcloud/bigtable/row_data.py]: You can hold off on the review here until #1494 is in if you like.

**Comment by dhermes** [comment]
@tseaver PTAL (I rebased on top of #1494)

**Comment by tseaver** [comment]
LGTM after the rebase/Travis.

**Pull request merged**